src/main/java/nl/mpcjanssen/simpletask/util/Util.kt

```diff
@@ -474,48 +475,77 @@ fun readAsset(assets: AssetManager, name: String): String {
     return buf.toString()
 }
 
-fun getRelativeThresholdDate(task: Task, ctx: Context): String? {
-    val date = task.thresholdDate
-    if (date != null) {
-        date.toDateTime()?.let {
-            return "T: " + RelativeDate.getRelativeDate(ctx, it)
-        }
-    }
-    return null
+fun getRelativeThresholdDate(task: Task, app: TodoApplication): String? {
+    val date = task.thresholdDate ?: return null
+    return getRelativeDate(app, "T: ", date).toString()
 }
 
-fun getRelativeDueDate(task: Task, application: TodoApplication, dueTodayColor: Int, overDueColor: Int, useColor: Boolean): SpannableString? {
-    val date = task.dueDate
-    if (date != null) {
-        date.toDateTime()?.let {
-            val relativeDate = RelativeDate.getRelativeDate(application, it)
-            val ss = SpannableString("Due: " + relativeDate)
-            if (date == application.today && useColor) {
-                setColor(ss, dueTodayColor)
-            } else if ((application.today.compareTo(date) > 0) && useColor) {
-                setColor(ss, overDueColor)
-            }
-            return ss
-        }
-    }
-    return null
+fun getRelativeDueDate(task: Task, app: TodoApplication): SpannableString? {
+    val date = task.dueDate ?: return null
+    return getRelativeDate(app, "Due: ", date)
 }
 
+/**
+ * This method returns a String representing the relative date by comparing
+ * the Calendar being passed in to the date / time that it is right now. It
+ * will compute both past and future relative dates. E.g., "one day ago" and
+ * "one day from now".
+ *
+
+ *
+ * **NOTE:** If the calendar date relative to "now" is older
+ * than one day, we display the actual date in its default format as
+ * specified by this class. If you don't want to
+ * show the actual date, but you want to show the relative date for days,
+ * months, and years, you can add the other cases in by copying the logic
+ * for hours, minutes, seconds.
+
+ * @param then date to calculate difference to
+ * *
+ * @return String representing the relative date
+ */
 
-private fun calculateRelativeAge(ctx: Context, dateString: String): String {
-    val date = dateString.toDateTime()
-    date?.let {
-        return RelativeDate.getRelativeDate(ctx, date)
```

Comment:
We had a wrapper in a wrapper here. We don't need two wrappers, so let's get rid of the middle one.